d2layouts/d2sequence/sequence_diagram.go

```diff
@@ -169,6 +226,34 @@ func (sd *sequenceDiagram) addLifelineEdges() {
 	}
 }
 
+func (sd *sequenceDiagram) placeNotes() {
+	rankToX := make(map[int]float64)
+	for _, actor := range sd.actors {
+		rankToX[sd.objectRank[actor]] = actor.Center().X
+	}
+
+	for _, note := range sd.notes {
+		verticalIndex := sd.verticalIndices[note.AbsID()]
+		y := sd.maxActorHeight + sd.yStep
+
+		for _, msg := range sd.messages {
+			if sd.verticalIndices[msg.AbsID()] < verticalIndex {
+				y += sd.yStep
+			}
+		}
+		for _, otherNote := range sd.notes {
```

Comment:
here you can start at the the current `note` index as the previous ones were already processed.
In the previous loop, you could keep a `lastEdgeIndex` outside the loop to avoid going through all of them again (though, it may hurt readability a bit, so up to you)